json_object.c

```diff
@@ -629,9 +642,18 @@ void json_object_object_del(struct json_object *jso, const char *key)
 static int json_object_boolean_to_json_string(struct json_object *jso, struct printbuf *pb,
                                               int level, int flags)
 {
+	int ret;
+
+	if (flags & JSON_C_TO_STRING_COLOR)
+		printbuf_strappend(pb,"\033[0;35m");
+
 	if (JC_BOOL(jso)->c_boolean)
-		return printbuf_strappend(pb, "true");
-	return printbuf_strappend(pb, "false");
+		ret = printbuf_strappend(pb, "true");
+	else
+		ret = printbuf_strappend(pb, "false");
+	if (flags & JSON_C_TO_STRING_COLOR)
+		return printbuf_strappend(pb,"\033[0m");
```

Comment:
If the previous printbuf_strappend() returned failure, we should not call this one.